### PR TITLE
OPS-966_Wire_Subit_Approval_Button

### PR DESCRIPTION
**Pull request by tdonaworth**
## What changed

- minor cleanup on API code (remove duplicate code)
- Added "In Review" styling to PreviewTable
- Added handleSubmitClick on Approve Agreement page
  - handles minor error checking; if no BLI are in DRAFT, button is disabled.
  - redirects back to `/agreements/` when complete
## Issue

#966

## How to test

Try to submit an Agreement that contains BLI with DRAFT status. Status of BLI should be updated to "In Review", should redirect to `/agreements/` list page.

## Screenshots

image

### CODE REVIEW

**Review by fpigeonjr — COMMENTED** [review]
tested locally and these were my observations.
1. In the Agreements list, when you expand the an item the `created_by` is set to a number instead of the name.SCR-20230510-nrgd

3. Created an agreement with a couple BLIs and when I went to submit for approval the button was disabled.SCR-20230510-nsru

File: frontend/src/pages/budgetLines/PreviewTable.scss [hunk 1]
```diff
@@ -3,3 +3,8 @@
 .usa-table th {
     padding: 0.5rem 0.5rem;
 }
+
+.underReview {
```
Comment (fpigeonjr) [inline frontend/src/pages/budgetLines/PreviewTable.scss:7]: maybe we can add this color to our global styles in `/sass/uswds/styles.scss` as suggested name:
`--data-viz-secondary-23: #8B4687`

**Comment by tdonaworth** [comment]
> tested locally and these were my observations.
> 
> 1. In the Agreements list, when you expand the an item the `created_by` is set to a number instead of the name.SCR-20230510-nrgd
> 2. Created an agreement with a couple BLIs and when I went to submit for approval the button was disabled.SCR-20230510-nsru

1. No clue what that is, as it wasn't a result of my changes.. does it happen in main?
2. I'll look into that.. worked during my testing, but perhaps there's some edge cases.

**Review by fpigeonjr — APPROVED** [review]
Tested locally and was able to submit the Agreement for review and see the tag update accordingly. 
However I am seeing the full_name not show up properly in the Agreement review page as shown here:
image

**Pull request merged**